src/Interpreters/Context.cpp

```diff
@@ -172,15 +172,42 @@ namespace ErrorCodes
 } while (false)                                         \
 
 
+/// Thread safe object initializer using double-checked locking
+class Initializer
```

Comment:
Looks ok. Maybe we can think about replacing it with `std::call_once`

Reply:
Why not something like

```
void initialize()
{
    static Initializer ...
}
```